app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java

```diff
@@ -95,6 +107,52 @@ public void afterTextChanged(final Editable s) {
         setEmojiCompatEnabled(false);
     }
 
+    @Override
+    protected void onFinishInflate() {
+        super.onFinishInflate();
+        defaultPaddingLeft = getPaddingLeft();
+    }
+
+    @Override
+    protected void onDraw(Canvas canvas) {
+        super.onDraw(canvas);
+
+        // If line numbers enabled and format is not .todo.txt
+        if (_nuEnabled && !(_hl instanceof TodoTxtSyntaxHighlighter)) {
+            drawLineNumbers(canvas);
+        } else if (getPaddingLeft() != defaultPaddingLeft) {
+            // Reset padding without line numbers fence
+            setPadding(defaultPaddingLeft, getPaddingTop(), getPaddingRight(), getPaddingBottom());
+        }
+    }
+
+    private void drawLineNumbers(Canvas canvas) {
+        // Draw line numbers
+        paint.setColor(Color.GRAY);
```

Comment:
Does this color work well in all cases? (Light vs dark etc).

iirc we use dark grey in some places for better readability,

Reply:
It can work well in dark mode, like this

![Screenshot_20230623_131738_net.gsantner.markor_test_edit_647072483566368.jpg](https://github.com/gsantner/markor/assets/60415467/f9f3e760-7de1-4ebc-9c46-71434e9c3a38)

Reply:
Looks nice! 😄